### ISSUE TITLE
Wizard returns a strange key (encoded ?)

### ISSUE THREAD

**Issue by AlainKaim** [TuyaAPI/cli#160]
Hello
From time to time, one of my device is disconnected, so I have to connect it again with the Tuya App. And then, I use tuya-cli wizard to get the new key and modify my homebridge config file.
This is usually a smooth process, but today, after successfully reconnecting a blind to Tuya app( and it works through it), the wizard returned a strange key : 
  {
    name: 'Volet salon petit',
    id: 'XXXXXXX',
    key: "1C<S<'HIBk*o_{eY"
  },

The name and id fields are ok. And the full informations are also ok for my other devices.
It is just this key which is odd, and of course unusable in homebridge.

Any help will be appreciated :)

**Comment by ferni7** [comment]
I've just came here to report the same thing, device disconnected had to re-add it with tuya app and now this one device has this weird key which doesn't work..

```
  {
    name: 'Stairs',
    id: 'xxxxxx',
    key: 'nKwYCKpsudKQn^?F'
  },
```

**Comment by caco3** [comment]
Note that it at least is consistent with what the Tuya IoT REST API provides:

https://eu.iot.tuya.com/cloud/explorer?groupId=group-home&interfaceId=470224448454738 > "Device Management" > "Query Device Details in Bulk" returns:
```
{
  "result": [
    {
      [..]
      "local_key": "V+IuH~QWR`)v6A{r",
      [..]
    }
  ],
  "success": true,
  "t": 1687381074266,
  "tid": "4acd3a94107611eea68636790d793901"
}
```

And `tuya-cli wizard` returned
```
  {
    name: 'Wi-Fi hub',
    id: 'bf641c545b9e1920da14vi',
    key: 'V+IuH~QWR`)v6A{r'
  },
```

**Comment by murtazabasrai** [comment]
I have having the same issue after re-adding the device to Tuya App.
Does anyone know how to get the correct key?

**Comment by caco3** [comment]
Are you sure it is now encrypted?
How did the yey look like before?
I use exactly the key that gets returned and it works fine.

**Comment by murtazabasrai** [comment]
Hi, I tried and the returned key is correct. It's just that the key format has changed. Earlier it used to be in HEX. Now it's having symbols. 

The issue of new key not working was in the homebridge Tuya plug-in that I was using which was validating the key as per the old format. I have found the fix to that so this new key format is not an issue.
Thanks!

**Comment by ferni7** [comment]
@murtazabasrai was that the Tuya Plugin https://github.com/iRayanKhan/homebridge-tuya#readme ? 

If so can you please share the fix?

**Comment by murtazabasrai** [comment]
Yep that's the one I'm using and here's the details of the fix that resolved the issue:
https://github.com/iRayanKhan/homebridge-tuya/issues/409#issuecomment-1545078314

**Comment by AlainKaim** [comment]
This fix is working for me, thank you !!!I guess this will be modified in the next release of tuyalan….Have a great day !Le 5 juil. 2023 à 06:58, Murtaza Basrai ***@***.***> a écrit :﻿
Yep that's the one I'm using and here's the details of the fix that resolved the issue:
iRayanKhan/homebridge-tuya#409 (comment)
